src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala

```diff
@@ -587,6 +587,16 @@ $DOCKERFILE_SIGIL
     // This task requires that the docker dependency stage have been run.
     dockerDependencyStage.value
 
+    // Ensure that the base image is up to date. This also works around an issue wherein the
+    // `docker build` command will fail to authenticate to a repository even when credentials are
+    // valid. The `docker pull` command doesn't exhibit this issue.
+    // TODO: We should invalidate the dependency hash if this image has changed.
```

Comment:
@jkinkead I've left a TODO here instead of doing this because I'm not quite sure how to get command output.  The easiest way to do this would be to run `docker images -q <base-image>`, which returns the image ID (or nothing if the image has never been pulled). We'd then run `docker pull ...`, and finally run `docker images ...` again. If the output is different, invalidate the hash.